### PR TITLE
PUD-1488  API: Add note BE

### PR DESCRIPTION
**Pull request by AndyNewton**
Add notes and return them on get recall.

### CODE REVIEW

File: src/main/resources/db/migration/V83__Add_recall_notes_table.sql [hunk 1]
```diff
@@ -0,0 +1,19 @@
+CREATE TABLE note (
+                                id UUID PRIMARY KEY,
+                                recall_id UUID NOT NULL REFERENCES recall (id),
+                                subject TEXT NOT NULL,
+                                details TEXT NOT NULL,
+                                index INTEGER NOT NULL,
+                                document_id UUID REFERENCES document (id),
+                                created_by_user_id UUID NOT NULL,
```
Comment (philsegal) [inline src/main/resources/db/migration/V83__Add_recall_notes_table.sql:8]: should reference users table

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/component/pact/provider/ManageRecallsUIPactTest.kt [hunk 2]
```diff
@@ -265,6 +244,15 @@ class ManagerRecallsUiAuthorizedPactTest : ManagerRecallsUiPactTestBase() {
...
 
+  @State(
+    "a recall with notes exists",
+  )
+  fun `a recall with notes exists`() {
+    `a user and a fully populated recall without documents exists`()
+    `a note exists`(1, matchedNoteId, matchedDocumentId)
+    `a note exists`(2)
+  }
+
```
Comment (philsegal) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/component/pact/provider/ManageRecallsUIPactTest.kt:255]: Do we have a need for this?  We cant delete notes so not sure of the use case

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/controller/DocumentController.kt [hunk 3]
```diff
@@ -161,7 +161,7 @@ class DocumentController(
     if (generateDocumentRequest.category.uploaded)
       throw WrongDocumentTypeException(generateDocumentRequest.category)
 
-    return when (generateDocumentRequest.category) {
+    return when (generateDocumentRequest.category) { // TODO: all the below service calls should be annotated @Transactional - in their respective service classes of course
```
[the remark this answers is not in the document]
Comment (AndyNewton) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/controller/DocumentController.kt:164]: Done: PUD-1500; added to x3 TODOs

File: src/main/resources/db/migration/V83__Add_recall_notes_table.sql (hunk 1 above)
Comment (AndyNewton) [reply src/main/resources/db/migration/V83__Add_recall_notes_table.sql:8]: Good one, thanks, on it 👍

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/component/pact/provider/ManageRecallsUIPactTest.kt (hunk 2 above)
Comment (philsegal) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/component/pact/provider/ManageRecallsUIPactTest.kt:255]: Does the UI not add a note to a recall, and verify the response? Maybe not as it just does a single call (e.g. POST, not POST and then get recall)

File: src/main/resources/db/migration/V83__Add_recall_notes_table.sql (hunk 1 above)
Comment (AndyNewton) [reply src/main/resources/db/migration/V83__Add_recall_notes_table.sql:8]: I think this has been missed also for `rescind_record.created_by_user_id` @philsegal ?  Per: https://mojdt.slack.com/archives/C0255MPKHHC/p1645177103323509?thread_ts=1645136931.777759&cid=C0255MPKHHC

**Review by philsegal — APPROVED** [review]

**Pull request merged**